### DISCUSSION TITLE
connect with 3rd party OpenAI compatible API

### DISCUSSION THREAD

**Discussion by eschmidbauer** [KudoAI/chatgpt.js#245] [Ideas]
vLLM supports an OpenAI-Compatible API. is it possible to configure this code to connect to another service other than OpenAI? 
I've tried modifying the code but i am getting errors.
![image](https://github.com/kudoai/chatgpt.js/assets/7139998/7757e9cb-6c6d-4c80-83b5-8036ffe72697)

**Comment by eschmidbauer** [comment]
basically just these two changes and it doesn't seem to work. 

![image](https://github.com/kudoai/chatgpt.js/assets/7139998/3b640e3e-15ae-4937-9db3-59bd788cbcfc)
![Screenshot 2023-07-17 at 12 28 54 PM](https://github.com/kudoai/chatgpt.js/assets/7139998/08d97144-4627-4173-a054-569115c32001)

**Comment by eschmidbauer** [comment]
sorry it's a private URL- i will update the `manifest.content_scripts.matches` file and test again

**Comment by eschmidbauer** [comment]
![Screenshot 2023-07-19 at 12 38 27 PM](https://github.com/kudoai/chatgpt.js/assets/7139998/ecd66c46-06c0-4433-8de8-73c9d93149be)
same issue

**Comment by eschmidbauer** [comment]
it's the same API spec as OpenAI's
I ended up running streamlit to implement a chat interface using the openai package

**Comment by thegodone** [comment]
can we extend current solution to be compatible with chatGPT via azure connection : https://learn.microsoft.com/en-us/azure/ai-services/openai/reference ?
